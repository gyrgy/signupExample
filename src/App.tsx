import React from 'react';
import { Container } from '@material-ui/core';

const App = () => {
  return (
    <Container>
      <div>App</div>
    </Container>
  );
};

export default App;
