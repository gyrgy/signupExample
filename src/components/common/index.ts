export { default as ButtonPrimary } from 'src/components/common/buttonPrimary';
export { default as Checkbox } from 'src/components/common/checkbox';
export { default as Content } from 'src/components/common/content';
export { default as Fade } from 'src/components/common/fade';
export { default as Form } from 'src/components/common/form';
export { default as InputField } from 'src/components/common/inputField';
export { default as TabPanel } from 'src/components/common/tabPanel';