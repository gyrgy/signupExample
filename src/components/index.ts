export { default as SignUpDone } from 'src/components/signUpDone';
export { default as SignUpPrivacyForm } from 'src/components/signUpPrivacyForm';
export { default as SignUpTabMenu } from 'src/components/signUpTabMenu';
export { default as SignUpUserForm } from 'src/components/signUpUserForm';
