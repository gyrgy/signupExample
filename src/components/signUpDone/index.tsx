import React from 'react';
import { Alert } from '@material-ui/lab';
import { signUpDoneTestID } from 'src/constants/testIDs';
import { t } from 'src/utils/i18n';

interface SignUpDoneProps {
  testID?: string;
}

const SignUpDone = (props: SignUpDoneProps) => {
  const { testID } = props;
  return (
    <div data-testid={testID || signUpDoneTestID}>
      <Alert severity="success">
        {t('doneForm')}
      </Alert>
    </div>
  );
};

export default SignUpDone;