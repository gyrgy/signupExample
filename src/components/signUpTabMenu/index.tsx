import React, { ChangeEvent } from 'react';
import { AppBar, Tab, Tabs } from '@material-ui/core';
import { t } from 'src/utils/i18n';
import { tabMenuTestID, tabMenuUserTabTestID, tabMenuPrivacyTabTestID, tabMenuDoneTabTestID } from 'src/constants/testIDs';

interface TabMenuProps {
  selectedIndex: number;
  onChange: (index: number) => void;
  testID?: string;
}

const SignUpTabMenu = (props: TabMenuProps) => {
  const { selectedIndex, onChange, testID } = props;

  const handleTabChange = (event: ChangeEvent<{}>, newValue: number) => {
    onChange(newValue);
  };

  return (
    <AppBar position="static" color="default">
      <Tabs
        value={selectedIndex}
        onChange={handleTabChange}
        indicatorColor="primary"
        textColor="primary"
        variant="fullWidth"
        aria-label="full width tabs"
        data-testid={testID || tabMenuTestID}
      >
        <Tab label={t('tabsData.user')} disabled={selectedIndex !== 0} data-testid={tabMenuUserTabTestID} />
        <Tab label={t('tabsData.privacy')} disabled={selectedIndex !== 1} data-testid={tabMenuPrivacyTabTestID} />
        <Tab label={t('tabsData.done')} disabled={selectedIndex !== 2} data-testid={tabMenuDoneTabTestID} />
      </Tabs>
    </AppBar>
  );
};

export default SignUpTabMenu;
