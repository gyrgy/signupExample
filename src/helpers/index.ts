export { privacyFormCheckboxesHandler } from 'src/helpers/privacyFormCheckboxesHandler';
export { renderWithStore } from 'src/helpers/renderWithStore';
export { userFormFieldsStateHandler } from 'src/helpers/userFormFieldsStateHandler';
export { userFormFieldsValidationHandler } from 'src/helpers/userFormFieldsValidationHandler';
export { userFormSubmitHandler } from 'src/helpers/userFormSubmitHandler';
