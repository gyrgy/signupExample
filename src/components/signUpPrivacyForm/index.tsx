import React, { ChangeEvent } from 'react';
import { ButtonPrimary, Checkbox, Form } from 'src/components/common';
import { PrivacyFormCheckboxes } from 'src/constants/enums';
import { t } from 'src/utils/i18n';
import { privacyFormTestID, privacyFormTrayCheckboxTestID, privacyFormOtherCheckboxTestID } from 'src/constants/testIDs';

interface PrivacyFormProps {
  trayProductUpdatesChecked: boolean;
  otherProductsUpdatesChecked: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  submitForm: () => void;
}

const PrivacyForm = (props: PrivacyFormProps) => {
  const {
    trayProductUpdatesChecked,
    otherProductsUpdatesChecked,
    onChange,
    submitForm
  } = props;

  return (
    <Form testID={privacyFormTestID}>
      <Checkbox
        name={PrivacyFormCheckboxes.TrayProductUpdatesCheckboxName}
        label={t('privacyForm.trayProductUpdatesCheckboxLabel')}
        checked={trayProductUpdatesChecked}
        onChange={onChange}
        testID={privacyFormTrayCheckboxTestID}
      />
      <Checkbox
        name={PrivacyFormCheckboxes.OtherProductsUpdatesCheckboxName}
        label={t('privacyForm.otherProductsUpdatesCheckboxLabel')}
        checked={otherProductsUpdatesChecked}
        onChange={onChange}
        testID={privacyFormOtherCheckboxTestID}
      />
      <ButtonPrimary label="submit" onClick={submitForm} />
    </Form>
  );
};

export default PrivacyForm;