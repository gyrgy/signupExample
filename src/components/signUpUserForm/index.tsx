import React, { ChangeEvent, FocusEvent } from 'react';
import { Form, InputField, ButtonPrimary } from 'src/components/common';
import { UserFormFieldNames } from 'src/constants/enums';
import { t } from 'src/utils/i18n';
import {
  userFormTestID,
  userFormNameFieldTestID,
  userFormRoleFieldTestID,
  userFormEmailFieldTestID,
  userFormPasswordFieldTestID
} from 'src/constants/testIDs';

interface SignUpUserFromProps {
  userNameFieldValue: string;
  userNameFieldError: boolean;
  userRoleFieldValue: string;
  userRoleFieldError: boolean;
  userEmailFieldValue: string;
  userEmailFieldError: boolean;
  userPasswordFieldValue: string;
  userPasswordFieldError: boolean;
  handleChange: (event: ChangeEvent<HTMLInputElement>) => void;
  handleBlur:
  (event: FocusEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
  handleSubmit: () => void;
  testID?: string;
}

const SignUpUserFrom = (props: SignUpUserFromProps) => {
  const {
    userNameFieldValue,
    userNameFieldError,
    userRoleFieldValue,
    userRoleFieldError,
    userEmailFieldValue,
    userEmailFieldError,
    userPasswordFieldValue,
    userPasswordFieldError,
    handleChange,
    handleBlur,
    handleSubmit,
  } = props;

  return (
    <Form testID={userFormTestID}>
      <InputField
        type="text"
        label={t('userForm.userNameFieldLabel')}
        name={UserFormFieldNames.Name}
        onChange={handleChange}
        onBlur={handleBlur}
        value={userNameFieldValue}
        required
        error={userNameFieldError}
        errorHelperText={userNameFieldError ? t('userForm.userNameFieldErrorHelperText') : ''}
        testID={userFormNameFieldTestID}
      />
      <InputField
        type="text"
        label={t('userForm.userRoleFieldLabel')}
        name={UserFormFieldNames.Role}
        onChange={handleChange}
        onBlur={handleBlur}
        value={userRoleFieldValue}
        error={userRoleFieldError}
        errorHelperText={userRoleFieldError ? t('userForm.userRoleFieldErrorHelperText') : ''}
        testID={userFormRoleFieldTestID}
      />
      <InputField
        type="text"
        label={t('userForm.userEmailFieldLabel')}
        name={UserFormFieldNames.Email}
        onChange={handleChange}
        onBlur={handleBlur}
        value={userEmailFieldValue}
        required
        error={userEmailFieldError}
        errorHelperText={userEmailFieldError ? t('userForm.userEmailFieldErrorHelperText') : ''}
        testID={userFormEmailFieldTestID}
      />
      <InputField
        type="password"
        label={t('userForm.userPasswordFieldLabel')}
        name={UserFormFieldNames.Password}
        onChange={handleChange}
        onBlur={handleBlur}
        value={userPasswordFieldValue}
        required
        error={userPasswordFieldError}
        errorHelperText={userPasswordFieldError ? t('userForm.userPasswordFieldErrorHelperText') : ''}
        testID={userFormPasswordFieldTestID}
      />
      <ButtonPrimary label="submit" onClick={handleSubmit} />
    </Form>
  );
};

export default SignUpUserFrom;
