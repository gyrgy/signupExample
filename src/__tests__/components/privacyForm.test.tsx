import React from 'react';
import {
  render,
  RenderResult,
  cleanup,
  fireEvent
} from '@testing-library/react';
import { SignUpPrivacyForm } from 'src/components';
import {
  privacyFormTestID,
  privacyFormTrayCheckboxTestID,
  privacyFormOtherCheckboxTestID,
  primaryButtonTestID
} from 'src/constants/testIDs';

describe('test tab panel component', () => {
  let trayProductUpdatesChecked = false;
  let otherProductsUpdatesChecked = false;
  const spyOnChangeHandler = jest.fn();
  const spyOnSubmitHandler = jest.fn();
  let form: RenderResult;

  beforeEach(() => {
    form = render(
      <SignUpPrivacyForm
        trayProductUpdatesChecked={trayProductUpdatesChecked}
        otherProductsUpdatesChecked={otherProductsUpdatesChecked}
        onChange={spyOnChangeHandler}
        submitForm={spyOnSubmitHandler}
      />
    );
  });

  afterEach(cleanup);

  it('renders privacy form and children', () => {
    const { getByTestId } = form;
    const privacyForm = getByTestId(privacyFormTestID);
    const trayInput = getByTestId(privacyFormTrayCheckboxTestID);
    const otherInput = getByTestId(privacyFormOtherCheckboxTestID);
    const button = getByTestId(primaryButtonTestID);

    expect(privacyForm).toBeInTheDocument();
    expect(trayInput).toBeInTheDocument();
    expect(otherInput).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });

  it('tests tray checkbox to be checked', () => {
    trayProductUpdatesChecked = true;
    const { rerender, getByTestId } = form;
    rerender(
      <SignUpPrivacyForm
        trayProductUpdatesChecked={trayProductUpdatesChecked}
        otherProductsUpdatesChecked={otherProductsUpdatesChecked}
        onChange={spyOnChangeHandler}
        submitForm={spyOnSubmitHandler}
      />
    );

    const trayInputField = getByTestId(privacyFormTrayCheckboxTestID)
      .querySelector('input[type="checkbox"]') as HTMLInputElement;
    const otherInputField = getByTestId(privacyFormOtherCheckboxTestID)
      .querySelector('input[type="checkbox"]') as HTMLInputElement;
    expect(trayInputField).toHaveProperty('checked', true);
    expect(otherInputField).toHaveProperty('checked', false);
  });

  it('tests other checkbox to be checked', () => {
    trayProductUpdatesChecked = false;
    otherProductsUpdatesChecked = true;
    const { rerender, getByTestId } = form;
    rerender(
      <SignUpPrivacyForm
        trayProductUpdatesChecked={trayProductUpdatesChecked}
        otherProductsUpdatesChecked={otherProductsUpdatesChecked}
        onChange={spyOnChangeHandler}
        submitForm={spyOnSubmitHandler}
      />
    );

    const trayInputField = getByTestId(privacyFormTrayCheckboxTestID)
      .querySelector('input[type="checkbox"]') as HTMLInputElement;
    const otherInputField = getByTestId(privacyFormOtherCheckboxTestID)
      .querySelector('input[type="checkbox"]') as HTMLInputElement;
    expect(trayInputField).toHaveProperty('checked', false);
    expect(otherInputField).toHaveProperty('checked', true);
  });

  it('test callbacks', () => {
    const { getByTestId } = form;
    const trayInputField = getByTestId(privacyFormTrayCheckboxTestID)
      .querySelector('input[type="checkbox"]') as HTMLInputElement;
    const otherInputField = getByTestId(privacyFormOtherCheckboxTestID)
      .querySelector('input[type="checkbox"]') as HTMLInputElement;
    const button = getByTestId(primaryButtonTestID);

    fireEvent.click(trayInputField);
    fireEvent.click(otherInputField);
    fireEvent.click(button);

    expect(spyOnChangeHandler).toBeCalledTimes(2);
    expect(spyOnSubmitHandler).toBeCalledTimes(1);
  });

  it('tests if matches with snapshot', () => {
    const { asFragment } = form;
    expect(asFragment()).toMatchSnapshot();
  });
});