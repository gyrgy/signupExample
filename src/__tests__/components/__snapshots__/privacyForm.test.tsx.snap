// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test tab panel component tests if matches with snapshot 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="makeStyles-root-429"
    data-testid="components/signUpPrivacyForm"
    novalidate=""
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-442 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        data-testid="components/signUpPrivacyForm/trayCheckbox"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-445"
            data-indeterminate="false"
            name="trayProductUpdatesCheckbox"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Receive updates about Tray.io product by email
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-442 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-443 Mui-checked MuiIconButton-colorPrimary"
        data-testid="components/signUpPrivacyForm/otherCheckbox"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input-445"
            data-indeterminate="false"
            name="otherProductsUpdatesCheckbox"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Receive communication by email for other products created by Tray.io team
      </span>
    </label>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="components/constants/buttonPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        submit
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
